templates/deployment.yaml

```diff
@@ -44,18 +44,28 @@ spec:
       - name: {{ include "ld-relay.name" . }}-volume
         {{- toYaml .Values.relay.volume.definition | nindent 8 }}
       {{- end }}
+      {{ $has_projected_volume := false }}
       {{- range .Values.relay.secrets }}
-          {{- if .volumePath }}
-            {{ $has_volumes = true }}
-      - name: {{ .volumeName }}
-        secret:
-          secretName: {{ .secretName }}
-          items:
-            - key: {{ .secretKey }}
-              path:  {{ .volumePath }}
-          {{- end }}
+        {{- if .volumePath }}
+          {{ $has_projected_volume = true }}
```

Comment:
Have to make a pass through this once to set this variable; otherwise, the sources block would be empty, which is invalid.